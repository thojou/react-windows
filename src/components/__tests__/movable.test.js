import React from 'react';
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Movable from "../movable";

describe('Movable', () => {
    beforeAll(() => {
        ReactDOM.createPortal = jest.fn((element, node) => {
            return element;
        })
    });

    afterEach(() => {
        ReactDOM.createPortal.mockClear();
    });

    it('should render', () => {
        const tree = renderer.create(<Movable />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});