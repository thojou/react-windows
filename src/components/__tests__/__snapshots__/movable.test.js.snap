// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movable should render 1`] = `
<div
  className="react-draggable"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "boxSizing": "border-box",
      "cursor": "auto",
      "display": "inline-block",
      "flexShrink": 0,
      "height": "auto",
      "left": 0,
      "maxHeight": 9007199254740991,
      "maxWidth": 9007199254740991,
      "minHeight": 37,
      "minWidth": 248,
      "position": "absolute",
      "top": 0,
      "touchAction": "none",
      "transform": "translate(NaNpx,NaNpx)",
      "userSelect": "auto",
      "width": "auto",
    }
  }
  transform={null}
>
  <div
    className="sc-bdVaJa kdUzPB"
  >
    <div
      className="sc-EHOje gZQqER"
      rows="auto 1fr"
    >
      <div
        className="sc-bZQynM sc-gzVnrw jvbilf window-draggable"
        name="title"
      >
        WindowName
      </div>
      <div
        className="sc-bZQynM sc-gzVnrw kABAhf window-draggable"
        name="action"
      >
        <div>
          <button
            aria-label="minimize"
            className="sc-htoDjs ePQcQL"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 352H48c-26.5 0-48 21.5-48 48v32c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-32c0-26.5-21.5-48-48-48z"
              />
            </svg>
          </button>
          <button
            aria-label="maximize"
            className="sc-htoDjs ePQcQL"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-16 160H64v-84c0-6.6 5.4-12 12-12h360c6.6 0 12 5.4 12 12v84z"
              />
            </svg>
          </button>
          <button
            aria-label="restore"
            className="sc-htoDjs ePQcQL"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 48v288c0 26.5-21.5 48-48 48h-48V176c0-44.1-35.9-80-80-80H128V48c0-26.5 21.5-48 48-48h288c26.5 0 48 21.5 48 48zM384 176v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48h288c26.5 0 48 21.5 48 48zm-68 28c0-6.6-5.4-12-12-12H76c-6.6 0-12 5.4-12 12v52h252v-52z"
              />
            </svg>
          </button>
          <button
            aria-label="close"
            className="sc-htoDjs ePQcQL"
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        className="sc-bZQynM kYOTYT"
        name="body"
      >
        <div
          className="sc-bwzfXH bYIxcP"
        />
      </div>
    </div>
  </div>
  <span>
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "row-resize",
          "height": "10px",
          "left": "0px",
          "position": "absolute",
          "top": "-5px",
          "userSelect": "none",
          "width": "100%",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "col-resize",
          "height": "100%",
          "position": "absolute",
          "right": "-5px",
          "top": "0px",
          "userSelect": "none",
          "width": "10px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "bottom": "-5px",
          "cursor": "row-resize",
          "height": "10px",
          "left": "0px",
          "position": "absolute",
          "userSelect": "none",
          "width": "100%",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "col-resize",
          "height": "100%",
          "left": "-5px",
          "position": "absolute",
          "top": "0px",
          "userSelect": "none",
          "width": "10px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "ne-resize",
          "height": "20px",
          "position": "absolute",
          "right": "-10px",
          "top": "-10px",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "bottom": "-10px",
          "cursor": "se-resize",
          "height": "20px",
          "position": "absolute",
          "right": "-10px",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "bottom": "-10px",
          "cursor": "sw-resize",
          "height": "20px",
          "left": "-10px",
          "position": "absolute",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
    <div
      className=""
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "nw-resize",
          "height": "20px",
          "left": "-10px",
          "position": "absolute",
          "top": "-10px",
          "userSelect": "none",
          "width": "20px",
        }
      }
    />
  </span>
</div>
`;
